pkg/kubelet/dockershim/helpers.go

```diff
@@ -380,11 +381,33 @@ func ensureSandboxImageExists(client libdocker.Interface, image string) error {
 		return nil
 	}
 	if !libdocker.IsImageNotFoundError(err) {
-		return fmt.Errorf("failed to inspect sandbox image %q: %v", image, err)
+		return fmt.Errorf("failed to inspect pod infra container image %q: %v", image, err)
+	}
+
+	// To support images in private registries, try to read docker config
+	authConfig := dockertypes.AuthConfig{}
+	keyring := &credentialprovider.BasicDockerKeyring{}
+	pwd, _ := os.Getwd()
+	paths := []string{"/.docker", "/root/.docker", pwd + "/.docker"}
```

Comment:
Instead of assuming `root`, you should get the HOME environment variable: `os.Getenv("HOME")`